### PR TITLE
Include namespace in watch start event

### PR DESCRIPTION
**Pull request by cmurphy**
A watch can be initiated with refinement parameters: an ID, a selector, or a namespace. Without this patch, when a watch was started for a namespace, even though the watch was correctly started for the correct namespace, the returned start message did not include the namespace, which made it impossible for the client to verify what it was watching. This change ensures the namespace is included in the event.

Tests are added for the unexported stream method to verify the start event, which is less complex to test than the exported Watch method.

https://github.com/rancher/rancher/issues/40196

### CODE REVIEW

**Review by KevinJoiner — CHANGES_REQUESTED** [review]

File: pkg/subscribe/watcher_test.go
Comment (MbolotSuse) [inline pkg/subscribe/watcher_test.go:14]: Nit: Might also be good to include test cases for the error scenarios (no schema/store doesn't support watches/access control denied request). However, you didn't change that code here, so I'm ok with not including that.

**Review by MbolotSuse — COMMENTED** [review]
No issues with the approach, LGTM after we fix the issue with the "receive on channel in test" stuff.

File: pkg/subscribe/watcher_test.go
Comment (cmurphy) [reply pkg/subscribe/watcher_test.go]: Added a select here for <- done. I don't think I can block on <-result, that needs to be in its own goroutine and we need something else to signal that the goroutine is finished.

File: pkg/subscribe/watcher_test.go
Comment (cmurphy) [reply pkg/subscribe/watcher_test.go:16]: Added some error test cases

File: pkg/subscribe/watcher_test.go
Comment (KevinJoiner) [inline pkg/subscribe/watcher_test.go:137]:
```suggestion
			var gotEvent types.APIEvent
			select {
			case gotEvent = <-result:
```
I was able to try this out, and the test still runs since the separate go routine is not running any code. It is just blocking as well.

Separately this should not deadlock since you are using a buffered channel. 
If someone were to edit the implementation for `ws.stream(...)` to send multiple events on start, then we will hit a deadlock waiting for `ws.stream(...)` to return.

**Review by KevinJoiner — CHANGES_REQUESTED** [review]

File: pkg/subscribe/watcher_test.go
Comment (cmurphy) [reply pkg/subscribe/watcher_test.go]: You're right, the buffered channel made this unnecessary. I had needed it when I was trying to make it work with an unbuffered channel. Updated.

**Review by KevinJoiner — APPROVED** [review]

File: pkg/subscribe/watcher_test.go
Comment (MbolotSuse) [reply pkg/subscribe/watcher_test.go]: As long as the test doesn't stall (ran on my end with `go test -race` and looked fine), no issue on my end. Will leave it up to @KevinJoiner to resolve since he opened the thread first (though he's already approved so I think he's ok with it).

**Review by MbolotSuse — APPROVED** [review]
LGTM

**Pull request merged**
